am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -13,127 +13,171 @@
 
+    private Map<JsonPointer, Map.Entry<Set<Permission>, SecurityClassification>> attributePermissionsForRole;
+
+    private String roleName;
+
     @BeforeEach
     void setUp() {
+        roleName = UUID.randomUUID().toString();
+    }
+
+    @Test
+    void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassificationsForSpecifiedRole() {
         Map.Entry<Set<Permission>, SecurityClassification> readPermission = new Pair<>(ImmutableSet.of(READ), PUBLIC);
-        Map.Entry<Set<Permission>, SecurityClassification> createPermission =
-            new Pair<>(ImmutableSet.of(CREATE), PUBLIC);
-        Map.Entry<Set<Permission>, SecurityClassification> updatePermission =
-            new Pair<>(ImmutableSet.of(UPDATE), PUBLIC);
-
-        Map<JsonPointer, Map.Entry<Set<Permission>, SecurityClassification>> attributePermissionsForRole =
-            ImmutableMap.of(
-                JsonPointer.valueOf("/child"), readPermission,
-                JsonPointer.valueOf("/parent/age"), createPermission,
-                JsonPointer.valueOf("/address/street/line1"), createPermission
-            );
-
-        importerService.grantDefaultPermission(
-            DefaultPermissionGrant.builder()
-                .roleName(ROLE_NAME)
-                .resourceDefinition(ResourceDefinition.builder()
-                    .serviceName(SERVICE_NAME)
-                    .resourceType(RESOURCE_TYPE)
-                    .resourceName(RESOURCE_NAME)
-                    .build())
-                .attributePermissions(attributePermissionsForRole)
-                .build());
-
-        Map<JsonPointer, Map.Entry<Set<Permission>, SecurityClassification>> attributePermissionsForOtherRole =
-            ImmutableMap.of(
-                JsonPointer.valueOf(""), updatePermission,
-                JsonPointer.valueOf("/address"), createPermission
-            );
+
+        attributePermissionsForRole = ImmutableMap.of(
+            JsonPointer.valueOf("/child"), readPermission,
+            JsonPointer.valueOf("/parent/age"), readPermission
+        );
+
+        importerService.addRole(roleName, IDAM, PUBLIC, ROLE_BASED);
 
         importerService.grantDefaultPermission(DefaultPermissionGrant.builder()
-            .roleName(OTHER_ROLE_NAME)
+            .roleName(roleName)
             .resourceDefinition(ResourceDefinition.builder()
                 .serviceName(SERVICE_NAME)
                 .resourceType(RESOURCE_TYPE)
                 .resourceName(RESOURCE_NAME)
                 .build())
-            .attributePermissions(attributePermissionsForOtherRole)
+            .attributePermissions(attributePermissionsForRole)
             .build());
+
+        RolePermissions rolePermissions =
+            service.getRolePermissions(buildResource(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME), roleName);
+
+        assertThat(rolePermissions.getPermissions())
```

Comment:
I have a feeling doing `isEqual(RolePermissions.builder()...build())` might be more readable. Maybe worth trying.